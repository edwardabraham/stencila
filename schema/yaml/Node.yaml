title: Node
category: other
status: stable
description: Union type for all types of nodes in this schema, including primitives and entities
# All the individual entity and primitive types are added to the `anyOf` property of this
# schema during the build to JSON Schema. This file just serves as a stub for status, description etc
