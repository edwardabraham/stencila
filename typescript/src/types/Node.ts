// Generated file; do not edit. See `../rust/schema-gen` crate.
            
import { Array } from './Array'
import { ArrayValidator } from './ArrayValidator'
import { Article } from './Article'
import { AudioObject } from './AudioObject'
import { BooleanValidator } from './BooleanValidator'
import { Brand } from './Brand'
import { Button } from './Button'
import { Call } from './Call'
import { CallArgument } from './CallArgument'
import { Cite } from './Cite'
import { CiteGroup } from './CiteGroup'
import { Claim } from './Claim'
import { CodeBlock } from './CodeBlock'
import { CodeChunk } from './CodeChunk'
import { CodeError } from './CodeError'
import { CodeExpression } from './CodeExpression'
import { CodeFragment } from './CodeFragment'
import { Collection } from './Collection'
import { Comment } from './Comment'
import { ConstantValidator } from './ConstantValidator'
import { ContactPoint } from './ContactPoint'
import { Cord } from './Cord'
import { CreativeWork } from './CreativeWork'
import { Datatable } from './Datatable'
import { DatatableColumn } from './DatatableColumn'
import { Date } from './Date'
import { DateTime } from './DateTime'
import { DateTimeValidator } from './DateTimeValidator'
import { DateValidator } from './DateValidator'
import { DefinedTerm } from './DefinedTerm'
import { Directory } from './Directory'
import { Division } from './Division'
import { Duration } from './Duration'
import { DurationValidator } from './DurationValidator'
import { Emphasis } from './Emphasis'
import { EnumValidator } from './EnumValidator'
import { Enumeration } from './Enumeration'
import { ExecutionDependant } from './ExecutionDependant'
import { ExecutionDependency } from './ExecutionDependency'
import { ExecutionDigest } from './ExecutionDigest'
import { ExecutionTag } from './ExecutionTag'
import { Figure } from './Figure'
import { File } from './File'
import { For } from './For'
import { Form } from './Form'
import { Function } from './Function'
import { Grant } from './Grant'
import { Heading } from './Heading'
import { If } from './If'
import { IfClause } from './IfClause'
import { ImageObject } from './ImageObject'
import { Include } from './Include'
import { Integer } from './Integer'
import { IntegerValidator } from './IntegerValidator'
import { Link } from './Link'
import { List } from './List'
import { ListItem } from './ListItem'
import { MathBlock } from './MathBlock'
import { MathFragment } from './MathFragment'
import { MediaObject } from './MediaObject'
import { MonetaryGrant } from './MonetaryGrant'
import { Note } from './Note'
import { NumberValidator } from './NumberValidator'
import { Object } from './Object'
import { Organization } from './Organization'
import { Paragraph } from './Paragraph'
import { Parameter } from './Parameter'
import { Periodical } from './Periodical'
import { Person } from './Person'
import { PostalAddress } from './PostalAddress'
import { Product } from './Product'
import { PropertyValue } from './PropertyValue'
import { PublicationIssue } from './PublicationIssue'
import { PublicationVolume } from './PublicationVolume'
import { Quote } from './Quote'
import { QuoteBlock } from './QuoteBlock'
import { Review } from './Review'
import { SoftwareApplication } from './SoftwareApplication'
import { SoftwareSourceCode } from './SoftwareSourceCode'
import { Span } from './Span'
import { Strikeout } from './Strikeout'
import { StringValidator } from './StringValidator'
import { Strong } from './Strong'
import { Subscript } from './Subscript'
import { Superscript } from './Superscript'
import { Table } from './Table'
import { TableCell } from './TableCell'
import { TableRow } from './TableRow'
import { Text } from './Text'
import { ThematicBreak } from './ThematicBreak'
import { Thing } from './Thing'
import { Time } from './Time'
import { TimeValidator } from './TimeValidator'
import { Timestamp } from './Timestamp'
import { TimestampValidator } from './TimestampValidator'
import { TupleValidator } from './TupleValidator'
import { Underline } from './Underline'
import { UnsignedInteger } from './UnsignedInteger'
import { Variable } from './Variable'
import { VideoObject } from './VideoObject'

// Union type for all types in this schema, including primitives and entities
export type Node =
  null |
  boolean |
  Integer |
  UnsignedInteger |
  number |
  string |
  Cord |
  Array |
  ArrayValidator |
  Article |
  AudioObject |
  BooleanValidator |
  Brand |
  Button |
  Call |
  CallArgument |
  Cite |
  CiteGroup |
  Claim |
  CodeBlock |
  CodeChunk |
  CodeError |
  CodeExpression |
  CodeFragment |
  Collection |
  Comment |
  ConstantValidator |
  ContactPoint |
  CreativeWork |
  Datatable |
  DatatableColumn |
  Date |
  DateTime |
  DateTimeValidator |
  DateValidator |
  DefinedTerm |
  Directory |
  Division |
  Duration |
  DurationValidator |
  Emphasis |
  EnumValidator |
  Enumeration |
  ExecutionDependant |
  ExecutionDependency |
  ExecutionDigest |
  ExecutionTag |
  Figure |
  File |
  For |
  Form |
  Function |
  Grant |
  Heading |
  If |
  IfClause |
  ImageObject |
  Include |
  IntegerValidator |
  Link |
  List |
  ListItem |
  MathBlock |
  MathFragment |
  MediaObject |
  MonetaryGrant |
  Note |
  NumberValidator |
  Organization |
  Paragraph |
  Parameter |
  Periodical |
  Person |
  PostalAddress |
  Product |
  PropertyValue |
  PublicationIssue |
  PublicationVolume |
  Quote |
  QuoteBlock |
  Review |
  SoftwareApplication |
  SoftwareSourceCode |
  Span |
  Strikeout |
  StringValidator |
  Strong |
  Subscript |
  Superscript |
  Table |
  TableCell |
  TableRow |
  Text |
  ThematicBreak |
  Thing |
  Time |
  TimeValidator |
  Timestamp |
  TimestampValidator |
  TupleValidator |
  Underline |
  Variable |
  VideoObject |
  Object;
