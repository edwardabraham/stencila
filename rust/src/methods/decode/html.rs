use eyre::{bail, Result};
use stencila_schema::Node;

/// Decode a HTML document to a `Node`
pub fn decode(_html: &str) -> Result<Node> {
    bail!("TODO")
}
