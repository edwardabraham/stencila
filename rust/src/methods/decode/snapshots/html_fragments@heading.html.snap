---
source: rust/src/methods/decode/html.rs
expression: "decode_fragment(&content, Options::default())"
input_file: fixtures/fragments/html/heading.html

---
[
  {
    "type": "Heading",
    "content": [
      "Title of document"
    ],
    "depth": 1
  },
  {
    "type": "Paragraph",
    "content": [
      "The h1 tag is reserved for single use by the title. So a heading of depth 1 uses h2 etc"
    ]
  },
  {
    "type": "Heading",
    "content": [
      "Heading 1"
    ],
    "depth": 1
  },
  {
    "type": "Heading",
    "content": [
      "Heading 2"
    ],
    "depth": 2
  },
  {
    "type": "Heading",
    "content": [
      "Heading 3"
    ],
    "depth": 3
  },
  {
    "type": "Heading",
    "content": [
      "Heading 4"
    ],
    "depth": 4
  },
  {
    "type": "Heading",
    "content": [
      "Heading 5"
    ],
    "depth": 5
  },
  {
    "type": "Heading",
    "content": [
      "Heading with an id"
    ],
    "depth": 1,
    "id": "heading-with-an-id"
  }
]
